ext/php5/comms_php.c

```diff
@@ -1,57 +1,74 @@
 #include "comms_php.h"
 
+#include <SAPI.h>
 #include <stdatomic.h>
 
 #include "arrays.h"
 #include "compat_string.h"
 #include "compatibility.h"
 #include "coms.h"
+#include "ddshared.h"
 #include "ddtrace.h"
 #include "logging.h"
 #include "mpack/mpack.h"
 
 ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
 
-static void _comms_convert_append(zval *item, size_t offset, void *context) {
-    struct curl_slist **list = context;
-    zval converted;
-    TSRMLS_FETCH();
+static void dd_append_header(struct curl_slist **list, const char *key, const char *val) {
+    /* The longest Agent header should be:
+     * Datadog-Container-Id: <64-char-hash>
+     * So 256 should give us plenty of wiggle room.
+     */
+    char header[256];
+    size_t len = snprintf(header, sizeof(header), "%s: %s", key, val);
+    if (len > 0 && len < sizeof(header)) {
```

Comment:
Nits: when doing `sizeof` on an object (not a type), I prefer dropping the parenthesis e.g. `sizeof header`. This because of the [grammar](https://en.cppreference.com/w/c/language/sizeof):

    sizeof( type )
    sizeof expression

So having the parenthesis on an expression is redundant and potentially confusing e.g. "Is `header` a type?"

Reply:
I mess this up all the time. Thanks for catching it! :)